keel-plugin/src/main/kotlin/com/netflix/spinnaker/keel/plugin/ResourceHandler.kt

```diff
@@ -12,78 +12,86 @@ import com.netflix.spinnaker.keel.api.randomUID
 import com.netflix.spinnaker.keel.diff.ResourceDiff
 import com.netflix.spinnaker.keel.events.Task
 import org.slf4j.Logger
+import org.slf4j.LoggerFactory
 
 /**
  * @param S the spec type.
  * @param R the resolved model type.
  *
  * If those two are the same, use [SimpleResourceHandler] instead.
  */
-interface ResourceHandler<S : ResourceSpec, R : Any> : KeelPlugin {
+abstract class ResourceHandler<S : ResourceSpec, R : Any>(
```

Comment:
I converted this to an abstract class so I could make default functionality private/protected and/or final